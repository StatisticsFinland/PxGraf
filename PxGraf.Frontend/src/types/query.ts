/* istanbul ignore file */

/* Query - variable queries*/

/**
 * Type for a query that contains @see {@link IVariableQuery} objects for each variable.
 * @property {{ [key: string]: IVariableQuery }} - Queries for each variable.
 */
export type Query = { [key: string]: IVariableQuery };

/**
 * Enum that defines different variable value filter types.
 */
export enum FilterType {
    Item = 'item',
    All = 'all',
    Top = 'top',
    From = 'from'
}

/**
 * Interface for variable query properties.
 * @property {IValueFilter} valueFilter - Reference to value filter.
 * @property {boolean} selectable - Selected selectable variable.
 * @property {any} virtualValueDefinitions - Virtual value definitions.
  */
export interface IVariableQuery {
    valueFilter: IValueFilter
    selectable: boolean
    virtualValueDefinitions
}

/**
 * Interface for value filter properties.
 * @property {FilterType} type - Selected filter type.
 * @property {string[] | string | number} query - The query for the filter.
 */
export interface IValueFilter {
    type: FilterType;
    query?: string[] | string | number
}

/**
 * Type for string supporting multiple languages.
 */
type MultiLanguageString = { [key: string]: string };

/* CubeQuery - meta data editions*/

/**
 * Interface for cube query properties.
 * @property {MultiLanguageString} chartHeaderEdit - The chart header edition.
 * @property {{ [key: string]: IVariableEditions }} variableQueries - The variable edition queries.
 */
export interface ICubeQuery {
    chartHeaderEdit?: MultiLanguageString;
    variableQueries: { [key: string]: IVariableEditions };
}

/**
 * Interface for variable editions.
 * @property {{ [key: string]: IVariableValueEditions }} valueEdits - The value editions.
 */
export interface IVariableEditions {
    valueEdits: {
        [valueCode: string]: IVariableValueEditions;
    };
}

/**
 * Interface for variable value editions.
 * @property {MultiLanguageString} nameEdit - The name edition.
 * @property {IContentComponentEdition} contentComponent - The content component edition containing source and unit.
  */
export interface IVariableValueEditions {
    nameEdit? : MultiLanguageString;
    contentComponent? : IContentComponentEdition;
}

/**
 * Interface for content component edition.
 * @property {MultiLanguageString} sourceEdit - Edition of the source.
 * @property {MultiLanguageString} unitEdit - Edition of the unit.
 */
interface IContentComponentEdition {
    sourceEdit? : MultiLanguageString;
    unitEdit? : MultiLanguageString;
}

/**
 * Interface that stores information about the query.
 * @property {number} size - The size of the query.
 * @property {number} sizeWarningLimit - The size warning limit of the query.
 * @property {number} maximumSupportedSize - The maximum supported size of the query.
 * @property {string[]} validVisualizations - Names of valid visualization types for of the query.
 * @property {{[key: string]: MultiLanguageString}} visualizationRejectionReasons - Reasons for why certain visualization types are rejected for the query.
 * @property {number} maximumHeaderLength - The maximum header length of the query.
 */
export interface IQueryInfo {
    size: number;
    sizeWarningLimit: number;
    maximumSupportedSize: number;
    validVisualizations: string[];
    visualizationRejectionReasons: {[key: string]: MultiLanguageString};
    maximumHeaderLength: number;
}