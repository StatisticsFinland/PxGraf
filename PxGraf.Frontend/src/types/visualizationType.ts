/* istanbul ignore file */

/**
 * Enum that defines different visualization types.
 */
export enum VisualizationType {
    VerticalBarChart = 'VerticalBarChart',
    GroupVerticalBarChart = 'GroupVerticalBarChart',
    StackedVerticalBarChart = 'StackedVerticalBarChart',
    PercentVerticalBarChart = 'PercentVerticalBarChart',
    HorizontalBarChart = 'HorizontalBarChart',
    GroupHorizontalBarChart = 'GroupHorizontalBarChart',
    StackedHorizontalBarChart = 'StackedHorizontalBarChart',
    PercentHorizontalBarChart = 'PercentHorizontalBarChart',
    PyramidChart = 'PyramidChart',
    PieChart = 'PieChart',
    LineChart = 'LineChart',
    ScatterPlot = 'ScatterPlot',
    Table = 'Table'
}