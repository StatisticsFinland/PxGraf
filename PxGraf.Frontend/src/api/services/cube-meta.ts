/* istanbul ignore file */

import ApiClient from "api/client";
import { useQuery } from "react-query";
import { ICubeMeta } from "types/cubeMeta";
import { defaultQueryOptions } from "utils/ApiHelpers";

/**
 * Interface for cube metadata result. Cube is a multi-dimensional result of Px file data filtered by a query.
 * @property {boolean} isLoading - Flag to indicate if the data is still loading.
 * @property {boolean} isError - Flag to indicate if an error occurred during loading.
 * @property {ICubeMeta} data - The cube meta data as an @see {@link ICubeMeta} object.
 */
export interface ICubeMetaResult {
    isLoading: boolean;
    isError: boolean;
    data: ICubeMeta;
}

const fetchCubeMeta = async(idStack: string[]): Promise<ICubeMeta> => {
    const client = new ApiClient();
    const url = `creation/cube-meta/${idStack.join('/')}`;

    return await client.getAsync(url);
}

export const useCubeMetaQuery = (idStack: string[]): ICubeMetaResult => {
    return useQuery(
        ['cube-meta', idStack],
        () => fetchCubeMeta(idStack),
        defaultQueryOptions
    );
}