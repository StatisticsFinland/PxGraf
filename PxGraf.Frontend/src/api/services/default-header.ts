/* istanbul ignore file */

import ApiClient from "api/client";
import { merge } from "lodash";
import { useQuery } from "react-query";
import { MultiLanguageString } from "types/multiLanguageString";
import { Query } from "types/query";
import { buildTableReference, defaultQueryOptions } from "utils/ApiHelpers";

/**
 * Interface for the chart's header result
 * @property {boolean} isLoading - Flag to indicate if the data is still loading.
 * @property {boolean} isError - Flag to indicate if an error occurred while retrieving the header.
 * @property {MultiLanguageString} data - The header data represented as a multi-language string.
 */
export interface IHeaderResult {
    isLoading: boolean;
    isError: boolean;
    data: MultiLanguageString;
}

const fetchDefaultHeader = async(idStack: string[], query: Query): Promise<MultiLanguageString> => {
    const client = new ApiClient();
    const requestBody = JSON.stringify(
        merge(
            { tableReference: buildTableReference(idStack) },
            { variableQueries: query }
        )
    );
  
    const url = 'creation/default-header';
    return await client.postAsync(url, requestBody);
}

export const useDefaultHeaderQuery = (idStack: string[], query: Query): IHeaderResult => {
    return useQuery(
        ['default-header', ...idStack, query],
        () => fetchDefaultHeader(idStack, query),
        {
            ...defaultQueryOptions,
            enabled: query != null,
        }
    );
}