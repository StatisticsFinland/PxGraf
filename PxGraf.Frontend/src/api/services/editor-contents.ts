/* istanbul ignore file */

import ApiClient from "api/client";
import { useQuery } from "react-query";
import { IEditorContentsResponse } from "types/editorContentsResponse";
import { buildCubeQuery, defaultQueryOptions, useDebounceState } from "utils/ApiHelpers";
import { ICubeQuery, Query } from "types/query";

/**
 * Interface for editor contents result. Editor contents are used by the editor to populate the settings for adjusting the visualization.
 * @property {boolean} isLoading - Flag to indicate if the data is still loading.
 * @property {boolean} isError - Flag to indicate if an error occurred during loading.
 * @property {IEditorContentsResponse} data - The editor contents as an @see {@link IEditorContentsResponse} object.
 */ 
export interface IEditorContentsResult {
    isLoading: boolean;
    isError: boolean;
    data: IEditorContentsResponse;
}

const fetchEditorContents = async (idStack: string[], query: Query, metaEdits: ICubeQuery): Promise<IEditorContentsResponse> => {
    const client = new ApiClient();
    const url = 'creation/editor-contents';
    const requestBody = JSON.stringify(buildCubeQuery(query, metaEdits, idStack));
    return await client.postAsync(url, requestBody);
}

export const useEditorContentsQuery = (idStack: string[], query: Query, cubeQuery: ICubeQuery): IEditorContentsResult => {
    [idStack, query, cubeQuery] = useDebounceState(1000, idStack, query, cubeQuery);
    const queryKey = ['editor-contents', ...idStack, query, cubeQuery];
    return useQuery(
        queryKey,
        () => fetchEditorContents(idStack, query, cubeQuery),
        {
            ...defaultQueryOptions,
            enabled: query != null,
            keepPreviousData: true,
        }
    );
}