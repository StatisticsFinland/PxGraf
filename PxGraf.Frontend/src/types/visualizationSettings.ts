/* istanbul ignore file */

import {VisualizationType} from './visualizationType';

/**
 * Interface for visualization settings
 * @property {string[]} defaultSelectableVariableCodes - Variable codes for default selectable variable values.
 * @property {boolean} pivotRequested - Flag to indicate if the user has requested manual pivoting.
 * @property {boolean} cutYAxis - Flag to indicate if the y-axis should be cut based on the values.
 * @property {string} multiselectableVariableCode - The variable code for the multiselectable variable if any.
 * @property {string[]} rowVariableCodes - Variable codes for the row variables.
 * @property {string[]} columnVariableCodes - Variable codes for the column variables.
 * @property {string} sorting - Selected sorting order.
 * @property {boolean} matchXLabelsToEnd - Flag to indicate if the x-axis labels should be matched to the end.
 * @property {number} markerSize - The size of scatter plot markers.
 * @property {VisualizationType} selectedVisualization - The selected visualization type.
 * @property {boolean} showDataPoints - Flag to indicate if data points should be shown in the visualization.
 */
export interface IVisualizationSettings {
    defaultSelectableVariableCodes?: { [key: string]: string[] };
    pivotRequested?: boolean;
    cutYAxis?: boolean;
    multiselectableVariableCode?: string;
    rowVariableCodes?: string[];
    columnVariableCodes?: string[];
    sorting?: string;
    matchXLabelsToEnd?: boolean;
    markerSize?: number;
    selectedVisualization?: VisualizationType;
    showDataPoints?: boolean;
}