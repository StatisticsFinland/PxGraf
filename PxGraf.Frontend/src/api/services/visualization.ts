/* istanbul ignore file */

import ApiClient from "api/client";
import { IQueryVisualizationResponse } from "@statisticsfinland/pxvisualizer";
import { useQuery } from "react-query";
import { ICubeQuery, Query } from "types/query";
import { IVisualizationSettings } from "types/visualizationSettings";

import { buildCubeQuery, defaultQueryOptions, useDebounceState } from "utils/ApiHelpers";

/**
 * Interface for a visualization result.
 * @property {boolean} isLoading - Flag to indicate if the data is still loading.
 * @property {boolean} isError - Flag to indicate if an error occurred during loading.
 * @property {IQueryVisualizationResponse} data - The visualization result represented as PxVisualizer @see {@link IQueryVisualizationResponse}.
 */
export interface IVisualizationResult {
    isLoading: boolean;
    isError: boolean;
    data: IQueryVisualizationResponse;
}

const fetchVisualization = async (
    idStack: string[],
    query: Query,
    metaEdits: ICubeQuery,
    language: string,
    selectedVisualization: string,
    visualizationSettings: IVisualizationSettings
): Promise<IQueryVisualizationResponse> => {

    const client = new ApiClient();

    const requestBody = JSON.stringify({
        query: buildCubeQuery(query, metaEdits, idStack),
        language: language,
        visualizationSettings: {
            ...visualizationSettings,
            selectedVisualization: selectedVisualization,
        }
    });

    const url = 'creation/visualization';
    return await client.postAsync(url, requestBody);
}

export const useVisualizationQuery = (
    idStack: string[],
    query: Query,
    cubeQuery: ICubeQuery,
    language: string,
    selectedVisualization: string,
    visualizationSettings: IVisualizationSettings
): IVisualizationResult => {
    [idStack, query, cubeQuery, language, selectedVisualization, visualizationSettings] = useDebounceState(1000, idStack, query, cubeQuery, language, selectedVisualization, visualizationSettings);
    const queryKey = ['chart', ...idStack, query, cubeQuery, language, selectedVisualization, visualizationSettings];

    const checkSettingsValidity = (selectedVisualization: string, settings: IVisualizationSettings) => {
        if (settings == null || selectedVisualization == null) return false;
        const requireSorting = ["horizontalBarChart", "groupHorizontalBarChart", "stackedHorizontalBarChart", "percentHorizontalBarChart", "pieChart"];
        if (requireSorting.includes(selectedVisualization) && settings.sorting == null) return false;
    }

    return useQuery(
        queryKey,
        () => fetchVisualization(idStack, query, cubeQuery, language, selectedVisualization, visualizationSettings),
        {
            ...defaultQueryOptions,
            enabled:
                query != null &&
                checkSettingsValidity(selectedVisualization, visualizationSettings),
            keepPreviousData: true,
        }
    );
}