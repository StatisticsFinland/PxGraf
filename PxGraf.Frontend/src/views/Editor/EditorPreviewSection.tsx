import React from 'react';
import { Box } from '@mui/material';
import { EditorContext } from 'contexts/editorContext';
import styled from 'styled-components';
import Preview from 'components/Preview/Preview';
import { IQueryInfo, Query } from 'types/query';
import { VisualizationType } from 'types/visualizationType';
import { IVisualizationSettings } from 'types/visualizationSettings';
import { useTranslation } from 'react-i18next';
import UiLanguageContext from 'contexts/uiLanguageContext';

const PreviewWrapper = styled(Box)`
    grid-area: 'preview';
    display: block;
    position: relative;
    grid-template-rows: auto 1fr;
    padding: 16px;
    gap: 16px;
`;

const GuideTextWrapper = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
`;

interface IEditorPreviewSectionProps {
    path: string[];
    query: Query;
    selectedVisualization: VisualizationType;
    visualizationSettings: IVisualizationSettings;
    queryInfo: IQueryInfo;

}

/**
 * Component for previewing the visualization. Used in the @see {@link Editor} component. The preview is rendered in the @see {@link Preview} component.
 * @param {string[]} path Path to the current table in the Px file system.
 * @param {Query} query Current query settings.
 * @param {VisualizationType} visualizationType Selected visualization type.
 * @param {IVisualizationSettings} visualizationSettings Selected visualization settings.
 * @param {IQueryInfo} queryInfo Information about the query.
 */
export const EditorPreviewSection: React.FC<IEditorPreviewSectionProps> = ({path, query, selectedVisualization, visualizationSettings, queryInfo}) => {
    const { t } = useTranslation();
    const {cubeQuery} = React.useContext(EditorContext);
    const {languageTab} = React.useContext(UiLanguageContext);
    if (queryInfo?.validVisualizations?.length === 0) {
        if (Object.keys(queryInfo?.visualizationRejectionReasons).length > 0) {
            // Selected filters cannot produce a visualization
            return (
                <GuideTextWrapper>
                    <span>{t("previewGuide.impossibleToVisualize")}</span>
                </GuideTextWrapper>
            );
        }
        // Not enough filters
        return (
            <GuideTextWrapper>
                <span>{t("previewGuide.tooSmallQuery")}</span>
            </GuideTextWrapper>
        );
    }
    return (
        <PreviewWrapper>
            <Preview
              path={path}
              query={query}
              language={languageTab}
              cubeQueryTextEdits={cubeQuery}
              selectedVisualization={selectedVisualization}
              visualizationSettings={visualizationSettings}
            />
        </PreviewWrapper>
    );
}

export default EditorPreviewSection;