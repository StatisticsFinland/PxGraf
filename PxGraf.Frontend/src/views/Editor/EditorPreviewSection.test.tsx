import React from 'react';
import EditorPreviewSection from "./EditorPreviewSection";
import { render } from "@testing-library/react";
import { IVisualizationResult } from "api/services/visualization";
import { EVisualizationType, EVariableType, ETimeVariableInterval } from "@statisticsfinland/pxvisualizer";
import { FilterType, ICubeQuery, Query } from "types/query";
import { IVisualizationSettings } from "types/visualizationSettings";
import { VisualizationType } from "types/visualizationType";
import serializer from "../../testUtils/stripHighchartsHashes";

jest.mock('envVars', () => ({
    PxGrafUrl: 'pxGrafUrl.fi/',
    PublicUrl: 'publicUrl.fi/'
}));

const mockVisualizationQueryResult: IVisualizationResult = {
    isLoading: false,
    isError: false,
    data: {
        data: [3107, 2383, 1555, 1839],
        missingDataInfo: {},
        dataNotes: {},
        metaData: [
            {
                code: "Tiedot",
                name: {
                    fi: "Tiedot",
                    sv: "Uppgifter",
                    en: "Information",
                },
                note: null,
                type: EVariableType.Content,
                values: [
                    {
                        code: "kulutus_t",
                        name: {
                            fi: "kulutus_t-fi",
                            sv: "kulutus_t-sv",
                            en: "kulutus_t-en",
                        },
                        note: null,
                        isSum: false,
                        contentComponent: {
                            unit: {
                                fi: "1000 t",
                                sv: "1000 t",
                                en: "1000 t",
                            },
                            source: {
                                fi: "PxGraf-fi",
                                sv: "PxGraf-sv",
                                en: "PxGraf-en",
                            },
                            numberOfDecimals: 0,
                            lastUpdated: "2023-02-01T06:00:00Z",
                        },
                    },
                ],
            },
            {
                code: "Vuosi",
                name: {
                    fi: "Vuosi",
                    sv: "Ã…r",
                    en: "Year",
                },
                note: null,
                type: EVariableType.Time,
                values: [
                    {
                        code: "2018",
                        name: {
                            fi: "2018",
                            sv: "2018",
                            en: "2018",
                        },
                        note: null,
                        isSum: false,
                        contentComponent: null,
                    },
                    {
                        code: "2019",
                        name: {
                            fi: "2019",
                            sv: "2019",
                            en: "2019",
                        },
                        note: null,
                        isSum: false,
                        contentComponent: null,
                    },
                    {
                        code: "2020",
                        name: {
                            fi: "2020",
                            sv: "2020",
                            en: "2020",
                        },
                        note: null,
                        isSum: false,
                        contentComponent: null,
                    },
                    {
                        code: "2021",
                        name: {
                            fi: "2021",
                            sv: "2021",
                            en: "2021",
                        },
                        note: null,
                        isSum: false,
                        contentComponent: null,
                    },
                ],
            },
        ],
        selectableVariableCodes: [],
        rowVariableCodes: [],
        columnVariableCodes: ["Vuosi"],
        header: {
            fi: "kulutus_t-fi 2018-2021",
            sv: "kulutus_t-sv 2018-2021",
            en: "kulutus_t-en 2018-2021",
        },
        visualizationSettings: {
            visualizationType: EVisualizationType.VerticalBarChart,
            timeVariableIntervals: ETimeVariableInterval.Year,
            defaultSelectableVariableCodes: null,
        },
        tableReference: { hierarchy: ["foo", "bar"], name:  "table" }
    }
};

const mockPath = [
    "foo",
    "bar",
    "table.px"
];
const mockQuery: Query = {
    Vuosi: {
        valueFilter: {
            type: FilterType.Top,
            query: 4
        },
        selectable: false,
        virtualValueDefinitions: null
    },
    Tiedot: {
        valueFilter: {
            type: FilterType.Item,
            query: [
                "kulutus_t"
            ]
        },
        selectable: false,
        virtualValueDefinitions: null
    }
};

const mockSelectedVisualization = 'table';
const mockVisualizationSettings: IVisualizationSettings = {
    rowVariableCodes: [],
    columnVariableCodes: [
        "Vuosi",
        "Tiedot"
    ]
};

const mockQueryInfo = {
    maximumHeaderLength: 120,
    maximumSupportedSize: 100,
    size: 5,
    sizeWarningLimit: 75,
    validVisualizations: ['VerticalBarChart', 'HorizontalBarChart'],
    visualizationRejectionReasons: {
        'pieChart': {
            'fi': 'huono kaavio'
        }
    }
};

const mockQueryInfoNoValidVisualizations = {
    maximumHeaderLength: 120,
    maximumSupportedSize: 100,
    size: 5,
    sizeWarningLimit: 75,
    validVisualizations: [],
    visualizationRejectionReasons: {
        'pieChart': {
            'fi': 'huono kaavio'
        }
    }
};

const mockQueryInfoNoRejectionsOrVisualizations = {
    maximumHeaderLength: 120,
    maximumSupportedSize: 100,
    size: 5,
    sizeWarningLimit: 75,
    validVisualizations: [],
    visualizationRejectionReasons: {}
};

jest.mock('react-i18next', () => ({
    ...jest.requireActual('react-i18next'),
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: () => new Promise(() => null),
            },
        };
    },
}));

jest.mock('api/services/visualization', () => ({
    ...jest.requireActual('api/services/visualization'),
    useVisualizationQuery: (
        idStack: string[],
        query: Query,
        cubeQuery: ICubeQuery,
        language: string,
        selectedVisualization: string,
        visualizationSettings: IVisualizationSettings) => {
        return mockVisualizationQueryResult;
    },
}));

function mockComponentMocker(name) {
    return function MockComponent(...args) {
        return (
            <pre data-testid={name}>
               args={JSON.stringify(args)}
            </pre>
        );
    };
}

jest.mock('@statisticsfinland/pxvisualizer', () => {
    const lib = jest.requireActual("@statisticsfinland/pxvisualizer");
    return {
        ...lib,
        Chart: mockComponentMocker('Chart')
    }
});

describe('Rendering test', () => {
    beforeAll(() => {
        expect.addSnapshotSerializer(serializer);
    });

    it('renders correctly', () => {
        const {asFragment} = render(
            <EditorPreviewSection
                path={mockPath}
                query={mockQuery}
                selectedVisualization={mockSelectedVisualization as VisualizationType}
                visualizationSettings={mockVisualizationSettings}
                queryInfo={mockQueryInfo}
            />);
        expect(asFragment()).toMatchSnapshot();
    });
    it('renders correctly with no valid visualizations', () => {
        const {asFragment} = render(
            <EditorPreviewSection
                path={mockPath}
                query={mockQuery}
                selectedVisualization={mockSelectedVisualization as VisualizationType}
                visualizationSettings={mockVisualizationSettings}
                queryInfo={mockQueryInfoNoValidVisualizations}
            />);
        expect(asFragment()).toMatchSnapshot();
    });
    it('renders correctly with no valid visualizations or rejected visualizations', () => {
        const {asFragment} = render(
            <EditorPreviewSection
                path={mockPath}
                query={mockQuery}
                selectedVisualization={mockSelectedVisualization as VisualizationType}
                visualizationSettings={mockVisualizationSettings}
                queryInfo={mockQueryInfoNoRejectionsOrVisualizations}
            />);
        expect(asFragment()).toMatchSnapshot();
    });
});