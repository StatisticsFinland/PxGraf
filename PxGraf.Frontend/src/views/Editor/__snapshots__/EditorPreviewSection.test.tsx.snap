// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering test renders correctly 1`] = `
<DocumentFragment>
  <div
  class="sc-gsFSXq jknzzI MuiBox-root css-0"
>
  <div
    class="sc-dcJsrY hLoHLO"
  >
    <button
      aria-label="tooltip.open: tooltip.visualizationSize"
      class="sc-aXZVg lSzEq"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorInfo MuiSvgIcon-fontSizeMedium css-1yq1soz-MuiSvgIcon-root"
        data-testid="InfoIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m1 15h-2v-6h2zm0-8h-2V7h2z"
        />
      </svg>
    </button>
    <div
      aria-label="tooltip.visualizationSize"
      class="MuiToggleButtonGroup-root css-bghv3y-MuiToggleButtonGroup-root"
      role="group"
    >
      <button
        aria-pressed="true"
        class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root Mui-selected MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButtonGroup-firstButton css-d9c359-MuiButtonBase-root-MuiToggleButton-root"
        tabindex="0"
        type="button"
        value="100%"
      >
        <b>
          100%
        </b>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-pressed="false"
        class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButtonGroup-middleButton css-d9c359-MuiButtonBase-root-MuiToggleButton-root"
        tabindex="0"
        type="button"
        value="1200px"
      >
        1200px
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-pressed="false"
        class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButtonGroup-middleButton css-d9c359-MuiButtonBase-root-MuiToggleButton-root"
        tabindex="0"
        type="button"
        value="992px"
      >
        992px
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-pressed="false"
        class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButtonGroup-middleButton css-d9c359-MuiButtonBase-root-MuiToggleButton-root"
        tabindex="0"
        type="button"
        value="768px"
      >
        768px
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-pressed="false"
        class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButtonGroup-middleButton css-d9c359-MuiButtonBase-root-MuiToggleButton-root"
        tabindex="0"
        type="button"
        value="576px"
      >
        576px
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-pressed="false"
        class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButtonGroup-middleButton css-d9c359-MuiButtonBase-root-MuiToggleButton-root"
        tabindex="0"
        type="button"
        value="390px"
      >
        390px
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-pressed="false"
        class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButtonGroup-middleButton css-d9c359-MuiButtonBase-root-MuiToggleButton-root"
        tabindex="0"
        type="button"
        value="360px"
      >
        360px
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-pressed="false"
        class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButtonGroup-lastButton css-d9c359-MuiButtonBase-root-MuiToggleButton-root"
        tabindex="0"
        type="button"
        value="320px"
      >
        320px
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <div
    class="MuiStack-root css-pl8gqh-MuiStack-root"
  />
  <div
    class="sc-iGgWBj fhbmWs tk-table"
  >
    <pre
      data-testid="Chart"
    >
      args=[{"locale":"","pxGraphData":{"data":[3107,2383,1555,1839],"missingDataInfo":{},"dataNotes":{},"metaData":[{"code":"Tiedot","name":{"fi":"Tiedot","sv":"Uppgifter","en":"Information"},"note":null,"type":"C","values":[{"code":"kulutus_t","name":{"fi":"kulutus_t-fi","sv":"kulutus_t-sv","en":"kulutus_t-en"},"note":null,"isSum":false,"contentComponent":{"unit":{"fi":"1000 t","sv":"1000 t","en":"1000 t"},"source":{"fi":"PxGraf-fi","sv":"PxGraf-sv","en":"PxGraf-en"},"numberOfDecimals":0,"lastUpdated":"2023-02-01T06:00:00Z"}}]},{"code":"Vuosi","name":{"fi":"Vuosi","sv":"Ã…r","en":"Year"},"note":null,"type":"T","values":[{"code":"2018","name":{"fi":"2018","sv":"2018","en":"2018"},"note":null,"isSum":false,"contentComponent":null},{"code":"2019","name":{"fi":"2019","sv":"2019","en":"2019"},"note":null,"isSum":false,"contentComponent":null},{"code":"2020","name":{"fi":"2020","sv":"2020","en":"2020"},"note":null,"isSum":false,"contentComponent":null},{"code":"2021","name":{"fi":"2021","sv":"2021","en":"2021"},"note":null,"isSum":false,"contentComponent":null}]}],"selectableVariableCodes":[],"rowVariableCodes":[],"columnVariableCodes":["Vuosi"],"header":{"fi":"kulutus_t-fi 2018-2021","sv":"kulutus_t-sv 2018-2021","en":"kulutus_t-en 2018-2021"},"visualizationSettings":{"visualizationType":"VerticalBarChart","timeVariableIntervals":"Year","defaultSelectableVariableCodes":null},"tableReference":{"hierarchy":["foo","bar"],"name":"table"}},"selectedVariableCodes":{}},{}]
    </pre>
  </div>
</div>
</DocumentFragment>
`;

exports[`Rendering test renders correctly with no valid visualizations 1`] = `
<DocumentFragment>
  <div
  class="sc-kAyceB dZVFsH"
>
  <span>
    previewGuide.impossibleToVisualize
  </span>
</div>
</DocumentFragment>
`;

exports[`Rendering test renders correctly with no valid visualizations or rejected visualizations 1`] = `
<DocumentFragment>
  <div
  class="sc-kAyceB dZVFsH"
>
  <span>
    previewGuide.tooSmallQuery
  </span>
</div>
</DocumentFragment>
`;
